src/parser/parsers/changelogs.py

```diff
@@ -11,27 +13,60 @@
 from .constants import OUTPUT_DIR
 
 
+
+class HTMLTagRemover(HTMLParser):
+    # https://www.slingacademy.com/article/python-ways-to-remove-html-tags-from-a-string#Using_HTMLParser
+    def __init__(self):
+        super().__init__()
+        self.result = []
+
+    def handle_data(self, data):
+        self.result.append(data)
+
+    def get_text(self):
+        return ''.join(self.result)
+
+
 class ChangelogParser:
     def __init__(self):
         self.CHANGELOGS_DIR = os.path.join(os.path.dirname(__file__), '../raw-changelogs/')
         self.OUTPUT_DIR = OUTPUT_DIR
-        self.OUTPUT_CHANGELOGS = self.OUTPUT_DIR+'/changelogs'
+        self.OUTPUT_CHANGELOGS = self.OUTPUT_DIR + '/changelogs'
         self.resources = self._get_resources()
 
-    def run_all(self):
-        changelogs_by_date = {}
+    # download rss feed from changelog forum and parse entries
+    def fetch_forum_changelogs(self):
+        print('Fetching Changelog RSS feed')
+        # Forum Link: https://forums.playdeadlock.com/forums/changelog.10/
+        f_url = 'https://forums.playdeadlock.com/forums/changelog.10/index.rss'
+        # fetches first page, last 20 entries date-wise
+        feed = feedparser.parse(f_url)
+        for entry in feed.entries:
+            # Dependent on thread title being in the format "mm-dd-yyyy Update"
+            date = entry.title.replace(' Update', '')
+            html_remover = HTMLTagRemover()
+            # Potential for more than one post, we just grab the first
+            html_remover.feed(entry.content[0].value)
+            self.changelogs_by_date[date] = self.run(date, html_remover.get_text())
+
+    def process_local_changelogs(self):
         files = [f for f in listdir(self.CHANGELOGS_DIR) if isfile(join(self.CHANGELOGS_DIR, f))]
         for file in files:
             date = file.replace('.txt', '')
-            changelog = self.run(date)
-            changelogs_by_date[date] = changelog
+            changelog = self.run(date, self._read_local_logs(date))
+            self.changelogs_by_date[date] = changelog
+
+    def run_all(self, fetch_rss=False):
+        self.changelogs_by_date = {}
+        if fetch_rss:
+            self.fetch_forum_changelogs()
+        self.process_local_changelogs()
```

Comment:
So to test the fetched logs, I deleted all raw-changelogs and ran the parser. It seems there is some limit to the fetching, since there are massive chunks of info missing

Reply:
Ahh, I think it limits it to 20 per page, I'll add a pagination loop

Reply:
So the rss feed will only show the latest 20, no paging.  It won't capture the old ones, but it'll at least catch everything moving forward :/